lib/sycamore/sycamore/query/execution/sycamore_executor.py

```diff
@@ -257,12 +258,24 @@ def get_code_string(self):
 """
         return result
 
+    def _write_metadata_to_trace_dir(self, plan: LogicalPlan, query_id: str):
+        assert self.trace_dir is not None, "Writing metadata requires trace_dir to be set"
+        path = os.path.join(self.trace_dir, f"{query_id}/metadata/")
+        os.makedirs(path, exist_ok=True)
+        with open(os.path.join(path, "query_plan.pickle"), "wb") as f:
```

Comment:
Could we store this as JSON, please? Pickle files are so unfriendly, and are brittle because they depend on a certain version of the Python classes being available. Given that we are already using PyDantic for query plans we should leverage it here instead of using pickle.

Reply:
I had some serialization issues with json references in the past but I think that was before the pydantic refactoring so maybe it works now, let me give it a shot